geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -126,6 +127,25 @@ private int getArrayIndex(int listIndex) {
     return listIndex;
   }
 
+  /**
+   * @param elementToInsert element to insert into the set
+   * @param referenceElement element to insert next to
+   * @param before true if inserting before reference element, false if it is after
+   * @param region the region this instance is store in
+   * @param key the name of the list to add to
+   * @return the number of elements in the list after the element is inserted
+   */
+  public int linsert(byte[] elementToInsert, byte[] referenceElement, boolean before,
+      Region<RedisKey, RedisData> region, RedisKey key) {
+    int index = elementInsert(elementToInsert, referenceElement, before);
+    if (index == -1) {
+      return index;
+    }
+    storeChanges(region, key, new InsertByteArray(elementToInsert, index));
+
+    return elementList.size();
```

Comment:
This method should be updated to use versioning in the Delta, similar to what was done for the APPEND command in GEODE-10108.